x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsPersister.java

```diff
@@ -255,17 +259,50 @@ public void persistCategoryDefinition(CategoryDefinition category, Supplier<Bool
      * Persist the quantiles (blocking)
      */
     public void persistQuantiles(Quantiles quantiles, Supplier<Boolean> shouldRetry) {
-        Persistable persistable = new Persistable(quantiles.getJobId(), quantiles, Quantiles.documentId(quantiles.getJobId()));
-        persistable.persist(AnomalyDetectorsIndex.jobStateIndexWriteAlias(), shouldRetry);
+        String quantilesDocId = Quantiles.documentId(quantiles.getJobId());
+
+        SearchRequest searchRequest =
+            new SearchRequest(AnomalyDetectorsIndex.jobStateIndexPattern())
+                .source(new SearchSourceBuilder().size(1).query(new IdsQueryBuilder().addIds(quantilesDocId)));
+        SearchResponse searchResponse = client.search(searchRequest).actionGet();
```

Comment:
I am not 100% sure about this. 

This means that if we fail to search for the ID, the job will be flagged as failed immediately and stop running. 

I wonder if we should do retries on the synchronous search as we do for the persistence?